modules/apps/frontend-js/frontend-js-components-web/src/main/java/com/liferay/frontend/js/components/web/internal/servlet/taglib/FrontendComponentsTopHeadDynamicInclude.java

```diff
@@ -74,6 +84,26 @@ protected void activate(Map<String, Object> properties) {
 				FFFrontendJSComponentsConfiguration.class, properties);
 	}
 
+	private void _createRuntimeFeatureFlag(
+		StringBundler sb, String featureFlagName, boolean validator) {
+
+		sb.append("Liferay.__FF__.");
+		sb.append(featureFlagName);
+		sb.append(" = ");
+		sb.append(validator);
+		sb.append(StringPool.SEMICOLON);
+	}
```

Comment:
I feel that think passing in `sb` is not ideal here, it needlessly breaks separation of concerns. If we wanted a helper private function, it should build a string internally and return a string.

I have no idea how to call this to be Brian-friendly, maybe `buildFeatureFlagJSGlobalVariable`? If it were me, I would avoid having this function entirely, and maybe pull out `Liferay.__FF__.` to a constant, to avoid duplication. But you are welcome to try :sweat_smile:

Reply:
> I feel that think passing in sb is not ideal here, it needlessly breaks separation of concerns. If we wanted a helper private function, it should build a string internally and return a string.

How to do it with sb? I think we will have two different string bundlers. How to manage it?

Reply:
> maybe pull out Liferay.__FF__. to a constant, to avoid duplication. But you are welcome to try sweat_smile

You mean defining a string constant? This is what I thought at first but I don't see this pattern often around DXP so, I preffered go with the most simple approach that it can be 😓

Reply:
> You mean defining a string constant? This is what I thought at first but I don't see this pattern often around DXP so, I preffered go with the most simple approach that it can be

If you search for `private static final String`, you'll find 14242 results :slightly_smiling_face:

Reply:
> How to do it with sb? I think we will have two different string bundlers. How to manage it?

If you decide to keep the private helper method, I was thinking something like this:

```java
private String _createRuntimeFeatureFlag(String featureFlagName, boolean validator) {
	StringBundler sb = new StringBundler(5);

	sb.append("Liferay.__FF__.");
	sb.append(featureFlagName);
	sb.append(" = ");
	sb.append(validator);
	sb.append(StringPool.SEMICOLON);

	return sb.toString();
}

@Override
public void include(
		HttpServletRequest httpServletRequest,
		HttpServletResponse httpServletResponse, String key)
	throws IOException {

	...

	sb.append(_createRuntimeFeatureFlag("something", true));

	...
}
```

Just like in JS, in Java methods create their own scope.

Reply:
https://github.com/liferay-frontend/liferay-portal/pull/2072/commits/57e3240b02f22e8b4841dd947a341ee363030e17

Thanks!